solidity/contracts/TokenStaking.sol

```diff
@@ -565,7 +564,7 @@ contract TokenStaking is Authorizations, StakeDelegatable {
     function transferStakeOwnership(address operator, address newOwner) public {
         require(msg.sender == operators[operator].owner, "Not authorized");
         operators[operator].owner = newOwner;
-        emit StakeOwnershipTransferred(operator, newOwner);
+        emit StakeDelegated(newOwner, operator);
```

Comment:
I'm not sure about using the same event in two flows. The previous `StakeOwnershipTransferred` event made here more sense to me. It may cause confusion because this function is not delegating the stake in a usual way. Why did we decide to change the previous approach?

Reply:
I think I was just overprotective in regard to the contract size.

Instead of:
- Filter out `StakeDelegated` events by owner field
- Filter out `StakeDelegated` events by operator field, checking more recent events to make sure the delegation ownership has not been transferred.

We can:
- Filter out `StakeDelegated` events by owner field
- Filter out `StakeOwnershipTransferred` events by operator field, checking more recent events to make sure the delegation ownership has not been transferred.

Reply:
See https://github.com/keep-network/keep-core/pull/1990/commits/681a8ba752ab5d5170d36337d43e10a0ea200f64 please.